torchdynamo/guards.py

```diff
@@ -52,6 +60,85 @@
 )
 
 
+class NNModuleChangeTrackerUtil:
+    original_register_parameter = None
+    original_add_module = None
+    original_load_state_dict = None
+    original_setattr = None
+
+    @classmethod
+    def register_patches_on_torch_nn_module(cls):
+        if getattr(
+            torch.nn.Module, "_NNModuleChangeTrackerUtil__dynamo_module_patch", True
+        ):
+            # register_parameter
+            torch.nn.Module.__dynamo_module_patch = False
+
+            original_register_parameter = torch.nn.Module.register_parameter
+
+            @functools.wraps(original_register_parameter)
+            def custom_register_parameter(self, key, value):
+                NNModuleChangeTrackerUtil.invalidate_module(self)
+                return original_register_parameter(self, key, value)
+
+            torch.nn.Module.register_parameter = custom_register_parameter
+
+            # add_module
+            original_add_module = torch.nn.Module.add_module
```

Comment:
Since these should be fixed.  Perhaps we should just read them once in global scope.

Reply:
This may get called multiple times, as `de_register_patches_on_torch_nn_module` can get invoked from `reset` or when we exit dynamo context, so it felt more correct to grab them here. No strong opinions.